syz-ci/manager.go

```diff
@@ -855,35 +863,92 @@ func (mgr *Manager) uploadCoverReport() error {
 	return nil
 }
 
+func (mgr *Manager) uploadCoverStat(fuzzingMinutes int) error {
+	if !mgr.managercfg.Cover || mgr.cfg.CoverPipelinePath == "" {
+		return nil
+	}
+
+	// Report generation consumes 40G RAM. Generate one at a time.
+	// TODO: remove it once #4585 (symbolization tuning) is closed
+	select {
+	case <-buildSem.WaitC():
+	case <-mgr.stop:
+		return nil
+	}
+	defer buildSem.Signal()
+
+	resp, err := mgr.httpGET("/cover?jsonl=1")
+	if err != nil {
+		return fmt.Errorf("failed to get /cover?json=1 report: %w", err)
+	}
+	defer resp.Body.Close()
+
+	curTime := time.Now()
+	pr, pw := io.Pipe()
+	defer pr.Close()
+	go func() {
+		decoder := json.NewDecoder(resp.Body)
+		for decoder.More() {
+			var covInfo cover.CoverageInfo
+			if err := decoder.Decode(&covInfo); err != nil {
+				pw.CloseWithError(fmt.Errorf("failed to decode CoverageInfo: %w", err))
+				return
+			}
+			if err := cover.WriteCIJSONLine(pw, covInfo, cover.CIDetails{
+				Version:        1,
+				Timestamp:      curTime.Format(time.RFC3339Nano),
+				FuzzingMinutes: fuzzingMinutes,
+				Arch:           mgr.lastBuild.Arch,
+				BuildID:        mgr.lastBuild.ID,
+				Manager:        mgr.name,
+				KernelRepo:     mgr.lastBuild.KernelRepo,
+				KernelBranch:   mgr.lastBuild.KernelBranch,
+				KernelCommit:   mgr.lastBuild.KernelCommit,
+			}); err != nil {
+				pw.CloseWithError(fmt.Errorf("failed to write CIJSONLine: %w", err))
+				return
+			}
+		}
+		pw.Close()
+	}()
+	fileName := fmt.Sprintf("%s-%s-%d-%d.jsonl",
+		mgr.name, curTime.Format(time.DateOnly),
+		curTime.Hour(), curTime.Minute())
+	err = mgr.uploadFile(mgr.cfg.CoverPipelinePath, fileName, pr, false)
+	if err != nil {
+		return fmt.Errorf("failed to uploadFileGCS(): %w", err)
+	}
+	return nil
+}
+
 func (mgr *Manager) uploadCorpus() error {
+	if mgr.cfg.CorpusUploadPath == "" {
+		return nil
+	}
 	f, err := os.Open(filepath.Join(mgr.workDir, "corpus.db"))
 	if err != nil {
 		return err
 	}
 	defer f.Close()
-	return mgr.uploadFile(mgr.cfg.CorpusUploadPath, mgr.name+"-corpus.db", f)
+	return mgr.uploadFile(mgr.cfg.CorpusUploadPath, mgr.name+"-corpus.db", f, true)
 }
 
-func (mgr *Manager) uploadFile(dstPath, name string, file io.Reader) error {
+func (mgr *Manager) uploadFile(dstPath, name string, file io.Reader, allowPublishing bool) error {
 	URL, err := url.Parse(dstPath)
 	if err != nil {
 		return fmt.Errorf("failed to parse upload path: %w", err)
 	}
 	URL.Path = path.Join(URL.Path, name)
 	URLStr := URL.String()
 	log.Logf(0, "uploading %v to %v", name, URLStr)
-	if strings.HasPrefix(URLStr, "gs://") {
-		return uploadFileGCS(strings.TrimPrefix(URLStr, "gs://"), file, mgr.cfg.PublishGCS)
-	}
-	if strings.HasPrefix(URLStr, "http://") ||
-		strings.HasPrefix(URLStr, "https://") {
+	if strings.HasPrefix(URLStr, "http") {
```

Comment:
Something happened with https.

Reply:
I missed that the new string is "http" rather than "http://". We should be fine.

Not sure if somebody may use a GCS bucket w/o "gs:" prefix and which starts with "http", e.g. "http_accessible_reports". Previously that was uploaded to GCS, now we will try to upload to http and fail.
But this may not be too important.

Reply:
Let me revert this part. I feel you're hesitating.

Reply:
Done.